documentation/adr/0001-recording-architecture-decisions.md

```diff
@@ -0,0 +1,79 @@
+# Recording Architecture Decisions
+
+- **Status:** Proposed
+- **Last Modified:** 2023-06-07
+- **Related Issue:** TODO
+- **Deciders:** TODO
+- **Tags:** ADRs
+
+## Context and Problem Statement
+
+Important decisions about the structure of a codebase, the tools or platforms adopted, and the scope or vision of a product are made all throughout a project's lifecycle. Often these decisions are made and changed blindly with little record of why that decision was made, making it difficult for other collaborators and even the original decision makers to understand the justification behind that decision in the future.
+
+_What is the best way to document key architectural decisions made within the project so that future contributors can understand the justification for those decisions?_
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- **Explicit:** Important decisions about the project architecture should be clear and unambiguous
+- **Documented:** There should be a written record describing which decisions are made
+- **Accessible:** Decision records should be centrally located, searchable, and accessible to non-technical audiences
+- **Portable:** The decision records should be able to travel with the code when it's forked, cloned, etc.
+- **Maintainable:** The system we adopt should be easy to maintain or amend as needed.
```

Comment:
Yeah this is a good question! Hmmm ultimately I was thinking these could be moved into the wiki which relates to your comment below:

> I might be tempted to vote for an alternative we haven't considered: putting the ADRs in a separate wiki platform (whatever we end up using for our wiki) rather than in the repo, but I could be persuaded either way. Let's discuss verbally sometime.

That would make them easier to edit for non-technical users. 

What's worth noting is that GitBook could allow us to achieve both of these goals with their [GitHub integration](https://docs.gitbook.com/product-tour/git-sync/enabling-github-sync) which syncs the content between a GitHub repo and GitBook itself and allows you to [specify a root path for the docs](https://docs.gitbook.com/product-tour/git-sync/content-configuration#root)

Reading through this list again, though, I do think there's some inherent tension between the `centrally located` condition in the Accessible criterion and `travel with the code when it's forked, cloned, etc.` in the Portable criterion. It could be helpful to think through which of these is the more important goal. and if there's some other way to link ADRs to the codebase that they are relevant to if we decide not to have a mono-repo.

Reply:
Yeah it does seem like those are in tension. I would personally lean towards `centrally located` over `travel with the code...`, but I could definitely be persuaded in another direction!

Reply:
Can you also add this part as a decision driver for this: 

> Should also be part of the public record, in the open, etc.

Reply:
Added this in [this commit](https://github.com/HHS/grants-api/pull/15/commits/0e4207784335a5e4e8a6da53ce7fa979b9ea075e)